File: Ref/31_Next_Permutation_question/Python/main1.py
from arr import arr
print(arr)

length = len(arr)
if length <= 2:
    print( arr.reverse())
pointer = length - 2
        
while pointer >= 0 and arr[pointer] >= arr[pointer + 1]:
    pointer -= 1
        
if pointer == -1:
    print(arr.reverse())
        
for x in range(length - 1, pointer, -1):
    if arr[pointer] < arr[x]:
        arr[pointer], arr[x] = arr[x], arr[pointer]
        break
        
arr[pointer + 1:] = reversed(arr[pointer + 1:])


print(arr)

File: Ref/31_Next_Permutation_question/JS/main1.js
import { arr } from "./arr.js";

const main = (arr) => {
  const length = arr.length;
  console.log(length);

  if (length <= 2) {
    return arr.reverse();
  }


  let pointer = length - 2;

  //find the first decreasing element from the end
  while (pointer >= 0 && arr[pointer] >= arr[pointer + 1]) {
    pointer--;
  }

  //if no such element is found, reverse the array
  if (pointer === -1) {
    return arr.reverse();
  }






}

console.log(main(arr))

